src/main/java/com/google/devtools/build/lib/query2/common/CommonQueryOptions.java

```diff
@@ -203,6 +205,17 @@ public LabelPrinter getLabelPrinter(
         : LabelPrinter.displayForm(mainRepoMapping);
   }
 
+  @Option(
+      name = "output_file",
+      defaultValue = "null",
```

Comment:
Are you sure? `""` actually produces an empty string, which then resolves to the workspace root because generally `path.resolve("").equals(path)`. Whereas `null` is handled specially by the options parser.